exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/AbstractMergedStatistic.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.physical.impl.statistics;
+
+import org.apache.drill.exec.vector.ValueVector;
+
+public abstract class AbstractMergedStatistic extends Statistic implements MergedStatistic {
+  @Override
+  public String getName() {
+    throw new UnsupportedOperationException("getName() not implemented");
+  }
+
+  @Override
+  public String getInput() {
+    throw new UnsupportedOperationException("getInput() not implemented");
+  }
+
+  @Override
+  public void merge(ValueVector input) {
+    throw new UnsupportedOperationException("merge() not implemented");
+  }
+
+  @Override
+  public Object getStat(String colName) {
+    throw new UnsupportedOperationException("getStat() not implemented");
+  }
+
+  @Override
+  public void setOutput(ValueVector output) {
+    throw new UnsupportedOperationException("getOutput() not implemented");
+  }
+
+  @Override
+  public void configure(Object configurations) {
```

Comment:
Changed the types to be specific. This config does not do hash lookups. However, for functions like `merge()` and `setOutput()` do hash lookups which might not be as expensive given the alternative (columns mappings for `MapVector` which needs to be recomputed for each incoming batch). Also, all the `MergedStatistic` and `StatisticsMergeBatch` contain a single map and given that both the incoming(no. of minor fragments) and outgoing batches(1) would not be many we can keep the code simple here.